pennylane/beta/tapes/qnode.py

```diff
@@ -383,11 +383,38 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+
+        if self.interface == "autograd":
+            # HOTFIX: to maintain compatibility with core, here we treat
+            # all inputs that do not explicitly specify `requires_grad=False`
+            # as trainable. This should be removed at some point, forcing users
+            # to specify `requires_grad=True` for trainable parameters.

```

Comment:
I think it would be quite helpful for future coders to rephrase "to maintain compatibility with core", since after this is merged, the code _will_ be core and this could be confusing? Maybe just say "hotfix to treat all inputs..."?

Reply:
Good point!

Reply:
I assume you have thought about it, but this could not be done in the parameter update function defined by the autograd tape, right?

Reply:
Yep. We actually want different behaviour between the autograd tape and the autograd QNode, to match their UI.

* If using an autograd tape, and we have a gate with a scalar parameter `0.6`, we _don't_ want to treat this as a differentiable parameter. We only want to consider `np.array(0.6, requires_grad=True)` as trainable.

* If we are using the QNode, and the user passes `0.6` as a positional QNode parameter, then we _do_ want to treat this as trainable, to match the existing core behaviour and to ensure all of our demos keep working. The hotfix here simply wraps all positional arguments with `requires_grad=True`, and pass this to the tape. (Note: positional parameters that explicitly have `requires_grad=False` will be skipped).